src/thor/route_action.cc

```diff
@@ -210,9 +210,9 @@ namespace valhalla {
       // location is a BREAK or if this is the last location
       if (destination->stoptype_ == Location::StopType::BREAK) {
         // Move origin back to the last break and collect the throughs
-        std::vector<PathLocation> throughs;
+        std::list<PathLocation> throughs;
         while(origin->stoptype_ != Location::StopType::BREAK) {
-          throughs.push_back(*origin);
+          throughs.push_front(*origin);
```

Comment:
this was the fix. we roll the origin back to the last breakpoint and keep the throughs along the way. the problem is we are rolling backward, so it reversed the order of the throughs. to fix this we just needed to push_front instead of back. pushing front on a vector means copies so i changed it to a list because there was no reason we needed random access iterators into it